src/gridfs-stream/upload.ts

```diff
@@ -448,61 +492,65 @@ function doWrite(_this: any, chunk: any, encoding: any, callback?: Callback) {
   return false;
 }
 
-function getWriteOptions(_this: any) {
-  var obj = {} as any;
-  if (_this.options.writeConcern) {
-    obj.w = _this.options.writeConcern.w;
-    obj.wtimeout = _this.options.writeConcern.wtimeout;
-    obj.j = _this.options.writeConcern.j;
+function getWriteOptions(self: GridFSBucketWriteStream) {
+  var obj: WriteConcernOptions = {};
+  if (self.bucket.s.options.writeConcern) {
```

Comment:
hm are you sure this didn't break anything? Are they indeed the same `options`?

Reply:
I don't think inheritance was happening correctly here. I added this to the class.

```
this.writeConcern = WriteConcern.fromOptions(options) || bucket.s.options.writeConcern;
```